YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs

```diff
@@ -6,61 +6,117 @@
+    private Vector3 playerAppearPosition;
+
     private void Start()
     {
-        // hide player in house
-        playerRend = PlayerInside.GetComponent<SpriteRenderer>();
-        UnshowPlayer();
-        endlevel = LevelSettings.GetActiveEndLevelController();
         // activate end level controller
+        endlevel = LevelSettings.GetActiveEndLevelController();
+        // Get player (this may return null if the player is inactive)
+        player = GameObject.FindGameObjectWithTag("Player");
+        // Get player renderer
+        playerRend = PlayerInside.GetComponent<SpriteRenderer>();
+        // Per default, we do not show the player in the window
+        UnshowPlayer ();
+        // Initialize the player spawn position
+        playerAppearPosition = gameObject.transform.position + new Vector3(0, -0.5f, 0);
 
+        // Set this point in time as the last execution time of update.
+        // This means, update is first performed at time Time.time + timeBetweenUpdates
+        lastTime = Time.time;
     }
 
-    /// <summary>
-    /// Check if someone knocks on our door
-    /// </summary>
-    /// <param name="other"> the other collider which can only be the player </param>
-    void OnTriggerEnter2D(Collider2D other)
+    // Tell the house whether the player is close or not
+    public void NotifyPlayerIsClose (bool IsClose)
     {
-        // we need to end the game only if player hits exit
-        // normally only the player will be able to get here
-        // so this is just a safety measure
-        // the collider attached to the infection trigger is hit first
-        // hence we have to get its "parent" the player
-        if (other.gameObject.GetComponentInParent<HumanBase>() != null)
-        {
-            if (other.gameObject.GetComponentInParent<HumanBase>().tag == "Player")
-            {
-                // in this level we only have to get home
-                if (LevelSettings.GetActiveSceneName() == "YouVsVirus_Levelgethome")
-                {
-                    //activate player in house, deactivate player, end game
-                    StartCoroutine(SetPlayerInHouse(other.gameObject.GetComponentInParent<HumanBase>().gameObject));
-                }
-                // in this level we have to get home and get toilet paper
-                else if (LevelSettings.GetActiveSceneName() == "YouVsVirus_Levelsupermarket" && other.gameObject.GetComponentInParent<Player>().hasToiletpaper)
-                {
-                    //activate player in house, deactivate player, end game
-                    StartCoroutine(SetPlayerInHouse(other.gameObject.GetComponentInParent<HumanBase>().gameObject));
-   
-                }
-            }
+        playerIsClose = IsClose;
+    }
+
+    
+    // Tell the house whether the player is inside or not
+    // and set the player inside.
+    public void NotifyPlayerInside (bool IsInsideFlag)
+    {
+        // TODO: Set the player actually inside
```

Comment:
What do you mean? Take the player himself?

Reply:
No, i ment that this routine should also be able to display the sprite of the mini player in the window.
Which it does now, i just forgot to delete that comment.